server/controller/auth.js

```diff
@@ -24,38 +24,45 @@ export default class authController {
    * if query fails send error response
    * if hash fails sends server error response
    */
-  static Signup(req, res) {
-    const error = validationResult(req);
-    util.errorCheck(error, res);
-    const {
-      username, email, password, firstname, lastname, othername, phonenumber, isadmin
-    } = req.body;
-    bcrypt.hash(password, 3)
-      .then((hashedpw) => {
-        const query = 'INSERT INTO users(username, email, password, firstname, lastname, othername, phonenumber, isadmin) VALUES($1, $2, $3, $4, $5, $6, $7, $8) RETURNING user_id, username, email, firstname, lastname, othername, phonenumber, isadmin';
-        const values = [username, email, hashedpw, firstname, lastname, othername, phonenumber, isadmin || false];
-        return databaseConnection.query(query, values)
-          .then((response) => {
-            console.log(response.rows[0]);
-            if (response) {
-              const token = jwt.sign(
-                { email: response.rows[0].email, userId: response.rows[0].user_id, isAdmin: response.rows[0].isadmin },
-                process.env.SECRET,
-                { expiresIn: '10h' }
-              );
-              return res.status(201).json({ data: [{ token, user: response.rows[0] }] });
+  static async Signup(req, res) {
+    try {
+      const error = validationResult(req);
+      const errormsg = await util.errorCheck(error, res);
+      if (errormsg) {
+        return false;
+      }
+      const {
+        username, email, password, firstname, lastname, othername, phonenumber, isadmin
+      } = req.body;
+      bcrypt.hash(password, 3)
+        .then(async (hashedpw) => {
+          const query = 'INSERT INTO users(username, email, password, firstname, lastname, othername, phonenumber, "isAdmin") VALUES($1, $2, $3, $4, $5, $6, $7, $8) RETURNING id, username, email, firstname, lastname, othername, phonenumber, "isAdmin"';
+          const values = [username, email, hashedpw, firstname, lastname, othername, phonenumber, isadmin || false];
+          try {
+            const response = await databaseConnection.query(query, values);
+            if (response.rows[0]) {
+              const token = jwt.sign({ email: response.rows[0].email, userId: response.rows[0].id, isAdmin: response.rows[0].isAdmin }, process.env.SECRET, { expiresIn: '10h' });
```

Comment:
try to abstract the parameters for `sign` method of `jwt`

Reply:
@innocentEdosa ☝️

Reply:
The rubric states that status code be included in response. but during our last presentation, the panelist gave feedback saying it isnt neccessary because it already being sent to the browser.
i would fix it back now though
would also try to handle the abstraction
Thanks for the review

Reply:
okay, if that is the case and you were given such feedback during the presentation, then that is fine by me.

Reply:
Approved

Reply:
Meanwhile are you having issues with the abstraction?